### PR TITLE
Implementation of Data Export Challenge

### PR DESCRIPTION
**Pull request by agrawalarpit14**
The implementation details are as follows:
- The data export component does pseudonymization only to fetch the orders of the user as only the orders are stored by using `email: (email ? email.replace(/[aeiou]/gi, '*') : undefined),` in `datacreator.js`.
- Counting to the above fact, the returned order data has no information of the original user.
- But, the order ID consists of `insecurity.hash(email).slice(0, 4)`. So, it can be matched to find the original orderer.

Please let me know your views and suggestions.
Thank You!!

Part Of Issue #660

### CODE REVIEW

File: test/e2e/dataExportSpec.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const config = require('config')
+
+describe('/#/privacy-security/data-export', () => {
+  describe('challenge "dataExportChallenge"', () => {
+    it('should be possible to register a user', () => {
+      browser.get('/#/register')
+      element.all(by.tagName('input')).get(1).sendKeys('admun@' + config.get('application.domain'))
```
Comment (J12934) [inline test/e2e/dataExportSpec.js]: This is quite hard to read when accessing the elements by their index.
Would be better to get them by their `id` attribute.

Comment (agrawalarpit14) [reply test/e2e/dataExportSpec.js]: Sure. I have used id's to reference the input fields in the latest commit.
Thank You!! for feedback✌🏻

Comment (J12934) [reply test/e2e/dataExportSpec.js]: Nice! Thank you for the fast change 👍

**Review by J12934 — APPROVED** [review]

File: test/e2e/dataExportSpec.js (hunk 1 above)
Comment (agrawalarpit14) [reply test/e2e/dataExportSpec.js]: Your Welcome!!😆😁

**Comment by agrawalarpit14** [comment]
Hi @J12934 Can you please help me in understanding the reason of failure of tests?

**Comment by J12934** [comment]
The linter is failing to to inconsistent code style.
You can run `npx standard --fix` fix the issues.

**Comment by agrawalarpit14** [comment]
> The linter is failing to to inconsistent code style.
> You can run `npx standard --fix` fix the issues.

Actually I did `standard --fix` already.
I tried `npx standard --fix` after your comment, but it makes no change.(the output is as follows)
```
➜  juice-shop git:(chlng_660) ✗ npx standard --fix
➜  juice-shop git:(chlng_660) ✗ npx standard
➜  juice-shop git:(chlng_660) ✗
```

**Comment by J12934** [comment]
Make sure you have the correct standard version installed.
If that doesn't work try to fix the errors shown in the travis log manually.

**Comment by agrawalarpit14** [comment]
> Make sure you have the correct standard version installed.
> If that doesn't work try to fix the errors shown in the travis log manually.

Sure I m on it. Will update you about the status.
Thank You!!✌🏻

**Comment by agrawalarpit14** [comment]
Hi @J12934 I still don't understand the full reason, but the failure of tests had something to do with me not pulling the latest changes before creating the commit. I have fixed the errors now. I began to understand it after reading the logging output as pointed by you.
Thank You!! for the help😁😁✌🏻✌🏻

**Comment by bkimminich** [comment]
:+1: for this challenge! The description, category and difficulty just need some tweaking, but I can do that post-merge! :100:

**Comment by agrawalarpit14** [comment]
> 👍 for this challenge! The description, category and difficulty just need some tweaking, but I can do that post-merge! 💯

Sure. Thanks!!😁😁

**Pull request merged**